File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        
        word, temp, prefix  = True, [], ""
        
        for x in s:
            if x != " ":
                prefix += x
            else:
                if prefix != "":
                    temp.append(prefix)
                    prefix = ""
        
        if prefix!="":
            temp.append(prefix)
        
        temp.reverse()
        
        return " ".join(temp)
                
                
                
        

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
       
        string temp = "";
        stack<string>st;
        for(int i=0; i<s.size();i++){
            char ch = s[i];
            if(ch == ' '){
                if(temp != "")
                    st.push(temp);
                temp = "";
            }
 
            else
             temp+=ch;
        }
 
        if(temp!=""){
            st.push(temp);
        }
        string ans = "";
 
        while(!st.empty()){
            ans+=(st.top()+" ");
            st.pop();
        }
        ans.pop_back();
 
        return ans;      
    }
};